### ISSUE TITLE
Cucucumber Expressions Ruby: ParameterType creation issue

### ISSUE THREAD

**Issue by luke-hill** [cucumber/cucumber-ruby#1235]
Auto-parameters `use_for_snippets` and `prefer_for_regexp_match` cannot be redefined when instantiating new `ParameterType` Object in RubyMine

## Summary

When instantiating the following object
```
ParameterType(
  name: 'chars',
  regexp: /([\w \-\u0091-\u00ff]+)/,
  type: String,
  transformer: ->(_) { _ },
  use_for_snippets: true,
  prefer_for_regexp_match: false,
)
```

The bottom two parameters are passed through as `nil` objects. Which then fail the instantiation checks.
Creating the object without these parameters works fine.

## Expected Behavior

The bottom two parameters are passed through as `true` and `false`
Creating the object without these parameters works fine.

## Current Behavior

The bottom two parameters are passed through as `nil` objects. Which then fail the instantiation checks.
Creating the object without these parameters works fine.

## Steps to Reproduce (for bugs)

See codeblock

## Your Environment

* Version used: Cucumber `3`, Cucumber Expressions `4.0.3` I'm aware there is a V5 but it's not auto included or permitted in Cucumber `3`
* Operating System and version: Win7

NB: I'm using a RubyMine interpreter, which currently has a few issues with Cucumber. This could be related!

**Comment by aslakhellesoy** [comment]
Can you try with cucumber-ruby 3.1.0 and change the `ParameterType` to:

```ruby
ParameterType(
  name: 'chars',
  regexp: /([\w \-\u0091-\u00ff]+)/
)
```

Is this still an issue?

**Comment by luke-hill** [comment]
On cucumber `3.0.2`
- The instantiation issue is non-existent when *not* passing in `snippets/prefer for match`.
- The instantiation crashes when not providing a `transformer`
- The instantiation works when not providing Object `type`
Will test `3.1.0` before Friday

**Comment by luke-hill** [comment]
This is fixed in Cucumber Expressions `5.0.4`
